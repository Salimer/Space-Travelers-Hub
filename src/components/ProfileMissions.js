import { ListGroup } from 'react-bootstrap';
import PropTypes from 'prop-types';

const ProfileMissions = (props) => {
  const { missions } = props;

  return (
    <div>
      <h3>My missions</h3>
      <ListGroup as="ul">
        {missions.map((mission) => (
          <ListGroup.Item as="li" key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
        ))}
      </ListGroup>
    </div>
  );
};

ProfileMissions.propTypes = {
  missions: PropTypes.arrayOf(Object).isRequired,
};

export default ProfileMissions;
