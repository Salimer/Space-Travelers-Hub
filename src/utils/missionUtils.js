const filterReserved = (missions) => missions.filter((mission) => mission.reserved);

export default filterReserved;
