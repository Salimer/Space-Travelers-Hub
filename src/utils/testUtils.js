import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

const withRouter = (component) => (
  <BrowserRouter>
    {component}
  </BrowserRouter>
);

const withProvider = (component) => (
  <Provider store={store}>
    {component}
  </Provider>
);

export { withProvider, withRouter };
