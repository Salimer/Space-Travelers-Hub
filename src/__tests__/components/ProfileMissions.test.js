import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { withProvider } from '../../utils/testUtils';
import ProfileMissions from '../../components/ProfileMissions';
import filterReserved from '../../utils/missionUtils';

const mockStore = configureStore([]);

describe('testing ProfileMissions component', () => {
  test('render', () => {
    const missions = [
      {
        mission_id: '1',
        mission_name: 'mission 1',
        description: 'mission 1 description',
        reserved: false,
      },
      {
        mission_id: '2',
        mission_name: 'mission 2',
        description: 'mission 2 description',
        reserved: true,
      },
    ];
    const store = mockStore({ missions });
    const filteredMissions = filterReserved(missions);
    const tree = renderer.create(
      withProvider(
        <ProfileMissions
          missions={filteredMissions}
        />,
      ), store,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
