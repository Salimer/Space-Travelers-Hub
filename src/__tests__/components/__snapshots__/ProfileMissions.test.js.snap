// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ProfileMissions component render 1`] = `
<div>
  <h3>
    My missions
  </h3>
  <ul
    className="list-group"
    onKeyDown={[Function]}
  >
    <li
      className="list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      mission 2
    </li>
  </ul>
</div>
`;
